src/stdlib_linalg.fypp

```diff
@@ -80,8 +88,93 @@ module stdlib_linalg
     #:endfor
   end interface outer_product
 
+
+  ! Check for squareness
+  interface is_square
+    !! version: experimental
+    !!
+    !! Checks if a matrix (rank-2 array) is square
+    !! ([Specification](../page/specs/stdlib_linalg.html#description_4))
+    #:for k1, t1 in RCI_KINDS_TYPES
+      module procedure is_square_${t1[0]}$${k1}$
+    #:endfor
+  end interface is_square
+
+
+  ! Check for diagonality
+  interface is_diagonal
+    !! version: experimental
+    !!
+    !! Checks if a matrix (rank-2 array) is diagonal
+    !! ([Specification](../page/specs/stdlib_linalg.html#description_5))
+    #:for k1, t1 in RCI_KINDS_TYPES
+      module procedure is_diagonal_${t1[0]}$${k1}$
+    #:endfor
+  end interface is_diagonal
+
+
+  ! Check for symmetry
+  interface is_symmetric
+    !! version: experimental
+    !!
+    !! Checks if a matrix (rank-2 array) is symmetric
+    !! ([Specification](../page/specs/stdlib_linalg.html#description_6))
+    #:for k1, t1 in RCI_KINDS_TYPES
+      module procedure is_symmetric_${t1[0]}$${k1}$
+    #:endfor
+  end interface is_symmetric
+
+
+  ! Check for skew-symmetry
+  interface is_skew_symmetric
+    !! version: experimental
+    !!
+    !! Checks if a matrix (rank-2 array) is skew-symmetric
+    !! ([Specification](../page/specs/stdlib_linalg.html#description_7))
+    #:for k1, t1 in RCI_KINDS_TYPES
+      module procedure is_skew_symmetric_${t1[0]}$${k1}$
+    #:endfor
+  end interface is_skew_symmetric
+
+
+  ! Check for Hermiticity
+  interface is_hermitian
+    !! version: experimental
+    !!
+    !! Checks if a matrix (rank-2 array) is Hermitian
+    !! ([Specification](../page/specs/stdlib_linalg.html#description_8))
+    #:for k1, t1 in RCI_KINDS_TYPES
+      module procedure is_hermitian_${t1[0]}$${k1}$
+    #:endfor
+  end interface is_hermitian
+
+
+  ! Check for triangularity
+  interface is_triangular
+    !! version: experimental
+    !!
+    !! Checks if a matrix (rank-2 array) is triangular
+    !! ([Specification](../page/specs/stdlib_linalg.html#description_9))
+    #:for k1, t1 in RCI_KINDS_TYPES
+      module procedure is_triangular_${t1[0]}$${k1}$
+    #:endfor
+  end interface is_triangular
+  
+
+  ! Check for matrix being Hessenberg
```

Comment:
```suggestion
  ! Check for Hessenbergness
```
I wonder is this way to form a quality out of a noun? I'm not a native English speaker, so feel free to reject this.

Reply:
I think your suggestion is very reasonable, as would be (to a lesser extent) "Hessenbergity" and probably a few others. I hesitated to use any one of them just because I could not find any established noun versions at all, where "triangularity" and friends were somewhat popular and readily searchable.
I am fine with leaving it as is, or changing it to your suggestion if we're okay being the ones to set the standard for matrix property nouns.